### ISSUE TITLE
App not working in certain fields

### ISSUE THREAD

**Issue by toohazy** [lochidev/TextToolsPro#16]
The text expansion feature doesn't work in some apps, I can't see anything in common between the affected apps, but I'm probably too not know what to look for.

Examples of apps where text expansion doesn't work:
- bravenewpipe: search bar
- clipstack: search bar
- official YouTube: search bar
- simple note
- pure writer

**Comment by lochidev** [comment]
yeah, the app only works on EditText elements, https://developer.android.com/reference/android/widget/EditText

I will keep an eye out to see if I can fix some stuff

**Comment by 00apm** [comment]
+1 this feature is essential
Your app is very useful! It would be great if this could be fixed

Other apps where text expansion doesn't work:
- chrome browser: search/url bar
- firefox browser: search/url bar
- obsidian.md: edit mode
- wikipedia mobile app: search bar, article editing mode

**Comment by lochidev** [comment]
@00apm @toohazy Unfortunately, the only way around this is by having a floating EditText widget. See: https://f-droid.org/en/packages/genius.mohammad.floating.stickies/

So I think the best thing I could do is to bake something like what the above app does, into my app. So basically whenever, text expansion doesn't work, you'll have to type in the floating window and the expanded text will automatically be copied to your clipboard so you can paste that in, say, the YouTube search bar.  Before I code this however I would like community feedback.

Obviously, you can use the above app or similar ones today itself, the only addition would be, 
=> Automatic copying of the expanded text to your clipboard, saving you an extra selection & touch.

Thank you!

**Comment by lochidev** [comment]
Personally, I think since, floating window apps already exist, I don't see any point of integrating that into my app, just to save on e click. So looking for feedback.

To clarify why text expansions don't work on some apps, that's because they don't emit accessibility events when text is typed on them. They are probably using custom rendering systems (games using OpenGL, apps using Skia)

**Comment by kranix0** [comment]
Unfortunately it doesn't work in Evernote, which is where I really need it. It does work in Markor, a simple markdown editor.

**Comment by kranix0** [comment]
I experimented with Texpand to see how it works.

It was able to perform expansions in problematic (for Expandroid) text fields in Evernote, YouTube, Firefox, Chrome, Obsidian and Wikipedia.

Texpand at times shows a floating picker/search widget when performing a replacement.

Interestingly in Chrome, when I press Menu > Share > copy URL, I get a message saying "Texpand pasted from your clipboard" - even thought I hadn't intentionally interacted with Texpand at all. 

I hope these clues can help to find a way for Expandroid to support many more apps.

**Comment by kranix0** [comment]
I've got it working in Evernote, albeit inconsistently and [the cursor often jumps to the top of the note](https://github.com/lochidev/Expandroid/issues/50#issuecomment-4272327776).

If it doesn't work in Texpand, we can probably assume that it's an edge case, requiring additional effort to fix. We as users should respectfully adjust our expectations for Expandroid accordingly.

The FAQ in the Texpand app has a list of incompatible apps:

Aquamail
Nine Email
Gmail
Evernote
Outlook
Edison mail
Spark
Samsung Notes
Termius
Termux
Firefox
Firefox Preview
Google Docs
Microsoft OneNote
Microsoft Word

I don't think Firefox Preview exists anymore, so this may be an old list. It mostly works for me in Evernote and Gmail. For Firefox (and presumably all browsers) it depends on the type of field on the page.

**Comment by lochidev** [comment]
@kranix0 Thank you, your work is very helpful! I'll look into this soon.

**Comment by amwalters** [comment]
I can confirm these fixes allow me to use the copilot prompt inside Outlook.  Looking forward to the new release.

**Comment by lochidev** [comment]
Hey guys! Try: https://github.com/lochidev/Expandroid/releases/tag/v7.3.1

Go easy on me though, lots of more work to be done.

Thanks!
